### ISSUE TITLE
Problem with <base> tag

### ISSUE THREAD

**Issue by qzminski** [bryanbraun/anchorjs#98]
Hey guys, first of all thank you for this great little tool! 

When using this script with `<base>` tag on the page the anchor links are pointing always to the root URL, for example `http://domain.tld/#my-headline` even if we are on a subpage like `http://domain.tld/subpage.html`. 

I am not sure how and if this should be fixed but here is a little snippet I'd like to share that fixes the links after they are generated:

```js
anchors.elements.forEach((el) => {
    el.querySelector('a.anchorjs-link').href = `${window.location.pathname}#${el.id}`;
});
```

**Comment by bryanbraun** [comment]
This is an interesting one. I hadn't even heard of the `<base>` tag before this, but I did some research and, [sure enough](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base#Hint), [this is an issue that comes up from time to time](https://stackoverflow.com/questions/8108836/make-anchor-links-refer-to-the-current-page-when-using-base).

I think a fix for this should be pretty straightforward: we query for the `<base>` tag and if we find one, we make our anchors absolute links instead of relative links. It should only be a couple lines of code.

**Comment by qzminski** [comment]
Sounds good to me, this would ensure we maintain backwards compatibility. Shall I prepare a pull request for this?

**Comment by bryanbraun** [comment]
@qzminski, yeah, if you've got some time then go for it!

**Comment by bryanbraun** [comment]
This has been merged! I hope to do a release that includes this fix once #101 and #102 are addressed.
